mtree/util.go

```diff
@@ -6,19 +6,11 @@ import (
 
 func HashesToBranchesArray(hashes []Hash) []BranchElement {
 	result := []BranchElement{}
-	for i := 0; i*2 < len(hashes); i++ {
-		// for anyone who is courious why i*2 + 1 comes before i * 2
-		// it's agreement between client side and contract side
-		if i*2+1 >= len(hashes) {
-			result = append(result,
-				BranchElementFromHash(
-					Hash(DagData{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}),
-					hashes[i*2]))
-		} else {
-			result = append(result,
-				BranchElementFromHash(
-					hashes[i*2+1], hashes[i*2]))
-		}
+	for i := 0; i < len(hashes); i++ {
+		result = append(result,
+			BranchElementFromHash(
+				Hash(DagData{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}),
+				hashes[i]))
```

Comment:
You always prepend 16 zero bytes to the hashes and they will be discarded when we convert it to hex. It means no data compact is used here. Is that what you want?
Another thing is to update the readme at https://github.com/tranvictor/ethashproof#merkle-audit-proof-encoding accordingly.